<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="map_file" default="$(find rotf3)/maps/arena.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- laod Rotf3 to parameter "robot_description", for controller_manager-->
  <include file="$(find rotf3)/launch/include/description.launch.xml">
  </include>
  
  <!-- Robot_state -->
	<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" respawn="false">
    <param name="publish_frequency" type="double" value="50.0" />
    <!-- <param name="tf_prefix" value=""/> -->
  </node>
  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL -->
  <include file="$(find rotf3)/launch/amcl.launch"/>

  <!-- move_base -->
  <include file="$(find rotf3)/launch/move_base.launch">
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- run get_odom.py (the main navigator script) -->
  <!-- <node pkg="rotf3" name="get_odom" type="get_odom.py" output="screen"/> -->

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find rotf3)/rviz/default.rviz"/>
  </group>
</launch>
